test/ectest.c

```diff
@@ -876,6 +876,23 @@ static void prime_field_tests(void)
         BN_free(scalar3);
     }
 
+    {
+        ECPARAMETERS *ecparameters;
+        EC_GROUP *group2;
+
+        ecparameters = EC_GROUP_get_ecparameters(group, NULL);
+        if (ecparameters == NULL)
+            ABORT;
+        group2 = EC_GROUP_new_from_ecparameters(ecparameters);
+        if (group2 == NULL)
+            ABORT;
+        if (EC_GROUP_cmp(group, group2, ctx))
+            ABORT;
+
+        EC_GROUP_free(group2);
+        ECPARAMETERS_free(ecparameters);
```

Comment:
This really has nothing to do with prime field testing. I would add this test as a separate function (e.g. `paramater_tests()` or something), and add a call to it after the `internal_curve_test()` call in `main`.